### PR TITLE
Add support for ConsumerRecord in receive_for_consumer

### PR DESCRIPTION
**Pull request by acco**
This pull request adds support for receiving consumer records in addition to consumer events. The main changes include:

1. Implemented a new `receive_for_consumer` function for handling consumer records.
2. Added type specifications for the `receive_for_consumer` function.
3. Created a new SQL query file `receive_consumer_records.sql` for fetching consumer records.
4. Updated the `Query` module to include the new SQL query for consumer records.

The new implementation for consumer records follows a similar pattern to the existing consumer events logic, including:
- Respecting batch size and max_ack_pending limits
- Updating record visibility and delivery information
- Transforming retrieved records into the appropriate format

These changes enable the system to process and manage consumer records alongside consumer events, enhancing the overall functionality of the consumer system.

### CODE REVIEW

**Comment by acco** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/119?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#120** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/120?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#119** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/119?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#118** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/118?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#117** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/117?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#116** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/116?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#115** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/115?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#114** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/114?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#113** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/113?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#112** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/112?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#111** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/111?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#110** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/110?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @acco and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/sequin/consumers/consumers.ex [hunk 1]
```diff
@@ -429,6 +431,48 @@ defmodule Sequin.Consumers do
     end
   end
 
+  # TODO: Populate data by joining against source tables
+  def receive_for_consumer(%{message_kind: :record} = consumer, opts) do
```
Comment (acco) [inline lib/sequin/consumers/consumers.ex:435]: the first version (no fetching source data) looks v similar to receiving for events

File: lib/sequin/consumers/receive_consumer_records.sql [hunk 2]
```diff
@@ -0,0 +1,44 @@
+with deliverable_records as (
```
Comment (acco) [inline lib/sequin/consumers/receive_consumer_records.sql:1]: v similar

**Comment by RTLS** [comment]
> This pull request adds support for receiving consumer records in addition to consumer events. The main changes include:
> 
> 1. Implemented a new `receive_for_consumer` function for handling consumer records.
> 2. Added type specifications for the `receive_for_consumer` function.
> 3. Created a new SQL query file `receive_consumer_records.sql` for fetching consumer records.
> 4. Updated the `Query` module to include the new SQL query for consumer records.
> 
> The new implementation for consumer records follows a similar pattern to the existing consumer events logic, including:
> 
> * Respecting batch size and max_ack_pending limits
> * Updating record visibility and delivery information
> * Transforming retrieved records into the appropriate format
> 
> These changes enable the system to process and manage consumer records alongside consumer events, enhancing the overall functionality of the consumer system.

what's your workflow for getting claude to generate this?

**Review by RTLS — APPROVED** [review]

**Pull request merged**